cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -1798,6 +1798,12 @@ Slice::CsGenerator::validateMetadata(const UnitPtr& u)
             const string msg = "'cs:namespace' is deprecated; use 'cs:identifier' to remap modules instead";
             p->unit()->warning(metadata->file(), metadata->line(), Deprecated, msg);
 
+            if (auto cont = dynamic_pointer_cast<Contained>(p); cont && cont->hasMetadata("cs:identifier"))
+            {
+                return "the 'cs:namespace' metadata cannot be used alongside 'cs:identifier' - both change the name of "
+                       "the mapped module";
```

Comment:
should be "mapped namespace"

Reply:
I also find the message inaccurate. We could support `java:package` on an outer module (or at file level) and `csharp:identifier` on an inner module, but we chose not to. What about simply:

```
'cs:namespace' and 'cs:identifier' cannot be used together
```

Reply:
I went with: `"A Slice element can only have one of 'cs:namespace' and 'cs:identifier' applied to it"`

It was pointed out that "cannot be used alongside" is vague. And I think "cannot be used together" is too.
This new message is more specific about the problem and how to fix it, but still simpler than my original message.